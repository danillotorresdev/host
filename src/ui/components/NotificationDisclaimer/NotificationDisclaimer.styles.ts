import { styled } from "@linaria/react";

export const NotificationWrapper = styled.div`
  padding: 10px;
  border-radius: 4px;
  margin-bottom: 10px;
  width: 80%;
  margin: 0 auto;
  margin-top: 20px;
`;
