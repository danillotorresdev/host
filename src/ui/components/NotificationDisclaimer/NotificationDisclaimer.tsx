import * as S from "@/ui/components/NotificationDisclaimer/NotificationDisclaimer.styles";

type NotificationDisclaimerProps = {
  message: string;
  type: "error" | "info";
};

const notificationColorsMap = {
  error: {
    backgroundColor: "#ffd6d6",
    color: "#cc0000",
  },
  info: {
    backgroundColor: "#f0f0f0",
    color: "#333333",
  },
};

export const NotificationDisclaimer = ({
  message,
  type,
}: NotificationDisclaimerProps) => (
  <S.NotificationWrapper style={notificationColorsMap[type]}>
    <p>{message}</p>
  </S.NotificationWrapper>
);
