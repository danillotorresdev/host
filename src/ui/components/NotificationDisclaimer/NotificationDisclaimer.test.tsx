import { render, screen } from "@/utils/settings/tests/utilities";
import { NotificationDisclaimer } from "./NotificationDisclaimer";
import { expect, it, describe } from "vitest";

describe("NotificationDisclaimer", () => {
  it("should render the message", () => {
    const { asFragment } = render(
      <NotificationDisclaimer message="Error message" type="error" />,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("Error message")).toBeDefined();
  });

  it("should render the message with the correct type", () => {
    render(<NotificationDisclaimer message="Error message" type="error" />);
    expect(screen.getByText("Error message")).toBeDefined();
  });

  it("should render the message with the correct type", () => {
    render(<NotificationDisclaimer message="Success message" type="info" />);
    expect(screen.getByText("Success message")).toBeDefined();
  });
});
