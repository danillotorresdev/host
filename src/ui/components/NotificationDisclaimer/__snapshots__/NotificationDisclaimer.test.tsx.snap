// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotificationDisclaimer > should render the message 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 n1uu6e1r"
    style="background-color: rgb(255, 214, 214); color: rgb(204, 0, 0);"
  >
    <p>
      Error message
    </p>
  </div>
</DocumentFragment>
`;
